### ISSUE TITLE
[4.2.11]: Inconsistent data between getLineItemById and getAllLineItemsByOrderId

### ISSUE THREAD

**Issue by Nellyaa** [craftcms/commerce#3253]
### Description

When you use the functions getLineItemById and getAllLineItemsByOrderId you get inconsistent snapshot data.
One is already json decoded, while the other returns the json as a string.
In craft\commerce\services\LineItems

### Steps to reproduce

1. Get a lineItem by using getLineItemById and look at the snapshot data
2. Get a lineItem by using getAllLineItemsByOrderId and look at the snapshot data

### Expected behavior

Both functions return the snapshot in the same manner

### Actual behavior

getLineItemById returns the data as string
getAllLineItemsByOrderId returns the data as array

### Craft CMS version

Craft Pro 4.3.7.1

### Craft Commerce version

4.2.11

### Installed plugins and versions

-

**Comment by jonleverrier** [comment]
@Nellyaa This might be a short term solution https://github.com/craftcms/commerce/issues/1684#issuecomment-1534434053

**Comment by Nellyaa** [comment]
> @Nellyaa This might be a short term solution [#1684 (comment)](https://github.com/craftcms/commerce/issues/1684#issuecomment-1534434053)

Thank you, this is what we're currently doing! 
I still think it should be changed in a future major version to be consistent though.

**Comment by nfourtythree** [comment]
Hi all

Thank you for reporting. We have pushed up a fix which makes the return from these two methods consistent.

This fix will be included in the next release of Commerce.

To get this early, change your `craftcms/commerce` requirement in your project's `composer.json` to:

```json
"require": {
  "craftcms/commerce": "dev-develop#2d0a70ba634b1184eb7b0e0ca40e209e3485196c as 4.3.1",
  "...": "..."
}
```

Then run `composer update`.

Thanks!

**Comment by lukeholder** [comment]
This is now fixed in 4.3.2
